### PR TITLE
DiskRP Swagger for 09/30

### PR DESCRIPTION
**Pull request by scottlzhang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by yungezz** [comment]
add waitforArmFeedback since new API version added

**Comment by yungezz** [comment]
hi @scottlzhang could you pls fix CI failure by following guideline https://aka.ms/ci-fix

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/ApprovePrivateEndpointConnection.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "myResourceGroup",
+    "api-version": "2020-09-30",
+    "diskAccessName": "myDiskAccess",
+    "privateEndpointConnectionName": "myPrivateEndpointConnection",
+    "privateEndpointConnection": {
+      "properties": {
+        "privateLinkServiceConnectionState ": {
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/ApprovePrivateEndpointConnection.json:10]: please remove space

**Comment by akning-ms** [comment]
@ARM reviewer to confirm 
For R4015 - NestedResourcesMustHaveListOperation  whether proxy only Resource is special case?
and R2062 - XmsResourceInPutResponse linter failure

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 2]
```diff
@@ -0,0 +1,3205 @@
...
+        },
+        "x-ms-examples": {
+          "Delete a private endpoint connection under a disk access resource.": {
+            "$ref": "./examples/DeleteAPrivateEndpointConnection.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{vmRestorePointName}/diskRestorePoints/{diskRestorePointName}": {
```
Comment (pilor) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1513]: Where are all the APIs related to restorePointCollections and restorePoints?  How does a user discover these deeply nested resources with LISTs and GETs at each level?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 3]
```diff
@@ -0,0 +1,3205 @@
...
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "description": "Properties of disk restore point"
+    },
+    "DiskRestorePointList": {
```
Comment (pilor) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:2934]: Where is this used?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 4]
```diff
@@ -0,0 +1,3205 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "List all possible private link resources under disk access resource.": {
+            "$ref": "./examples/GetDiskAccessPrivateLinkResources.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskAccesses/{diskAccessName}/privateEndpointConnections/{privateEndpointConnectionName}": {
```
[the remark this answers is not in the document]
Comment (scottlzhang) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1362]: Added the LIST operation on line 1513 of disk.json

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 3 above)
Comment (scottlzhang) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:3020]: This is for LIST operation for DiskRestorePoints. Added LIST operation on line 1603 in disk.json

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 2 above)
Comment (scottlzhang) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1555]: restorePointCollections and restorePoints are created by CRP via privilege calls. The API for createing restorePoints(AllocateRestorePoints) will return the diskRestorePoint armId in response body to client.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 5]
```diff
@@ -0,0 +1,3303 @@
...
+          "type": "array",
+          "description": "Array of private link resources",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        }
+      },
+      "description": "A list of private link resources"
+    },
+    "PrivateEndpointConnectionListResult": {
```
Comment (mentat9) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:3111]: You should consider making this schema pageable by annotating it with x-ms-pageable and adding a nextLink property. That way, even if you don't implement it now, you can implement it later without a breaking change.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 2 above)
Comment (mentat9) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1555]: What about after creation time? Are there collection gets for restorePointCollections and  restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

Comment (mentat9) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1555]: What about after creation time? Are there collection gets for restorePointCollections and restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

Comment (mentat9) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1555]: What about after creation time? Are there collection gets for restorePointCollections and restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 5 above)
Comment (scottlzhang) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:3117]: Good point. Added pageable in line 1553 for list PrivateEndpointConnection and in line 1649 for list DiskRestorePoints

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 2 above)
Comment (scottlzhang) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:1558]: After creation time, CRP has APIs that return restorePointCollections/restorePoints here link below. 

https://msazure.visualstudio.com/One/_git/Compute-CPlat-Core?path=%2Fsrc%2FCRP%2FDev%2FComputeResourceProvider%2FService%2FAPI%2FControllers%2FRestorePointCollectionsControllerBase.cs&version=GBmaster&line=34&lineEnd=39&lineStartColumn=9&lineEndColumn=79&lineStyle=plain&_a=contents

**Review by heaths — COMMENTED** [review]
Key Vault reformatting looks fine. We have a PR out that should fix that for everyone that should be merged soon. /cc @herveyw-msft @Lusitanian (I keep seeing these same three files reformatted over and over again - good reason to get that PR in sooner than later 😉).

**Comment by yungezz** [comment]
hi @msyyc could you pls help to confirm impact to Python SDK track1 and track2? thanks.

**Comment by yungezz** [comment]
hi @ArcturusZhang could you pls help to look at the go SDK generation failure? thanks

**Comment by yungezz** [comment]
> hi @ArcturusZhang could you pls help to look at the go SDK generation failure? thanks

For GO SDK failure, it's caused by readme validation `the output-folder of a management plane package 'profile-hybrid-2020-09-01' must be in this pattern: '^services(/preview)?/[^/]+/mgmt/[^/]+/[^/]+$'`. Seems not caused by this PR. But need to be fixed. @scottlzhang

**Review by yungezz — APPROVED** [review]

**Comment by Sandido** [comment]
@yungezz , this PR re-introduced the preview version of the Gallery API, 2020-09-30. 
This version [had been reverted](https://github.com/Azure/azure-rest-api-specs/pull/12222/files#diff-5871f1ff4953a3d11f8779e01460fe0ae3cb27630c3868c55a53417859e83a0aL222) and should not have been re-introduced.

**Pull request merged**